### PR TITLE
Increase the performance of video fetching, some refactoring

### PR DESCRIPTION
**Pull request by ardaasln**
Every video fetching process was trying to reach to the file system to get token. Instead, token fetching is done only once at the initialization stage.

### CODE REVIEW

File: src/background/init.js [hunk 1]
```diff
@@ -39,6 +39,9 @@ function backgroundTask() {
       crawler.crawl(mainUrl, callback);
     },
   ], (err, results) => {
+    if (err) {
```
Comment (elhanarinc) [inline src/background/init.js:42]: If there is an error, return statement should be executed.
```
console.log(err.stack);
return err;
```

File: src/lib/youtube.js [hunk 2]
```diff
@@ -17,27 +18,20 @@ let TOKEN_PATH = TOKEN_DIR + 'google-apis-nodejs-quickstart.json';
...
 
   // Check if we have previously stored a token.
   fs.readFile(TOKEN_PATH, (err, token) => {
     if (err) {
-      if (typeof requestData !== 'function') {
-        getNewToken(oauth2Client, requestData, callback, callback2);
-      } else {
-        getNewToken(oauth2Client, requestData);
-      }
+      getNewToken(callback);
```
Comment (elhanarinc) [inline src/lib/youtube.js]: Again, there should be a return statement with callback.
```
return callback;
```

File: src/lib/youtube.js [hunk 3]
```diff
@@ -155,10 +145,14 @@ function createResource(properties) {
   return resource;
 }
 
-function videosListById(auth, requestData, callback) {
+function getVideoById(requestData, callback) {
+  if (typeof oauth2Client == 'undefined') {
```
Comment (elhanarinc) [inline src/lib/youtube.js]: Use [lodash](https://lodash.com/docs/) for this type of checks.

**Review by elhanarinc — CHANGES_REQUESTED** [review]
Add return statements for error cases.
Use lodash (older name is underscore) for handling objects.

**Comment by ardaasln** [comment]
Done with review hotfixes @elhanarinc

**Review by elhanarinc — APPROVED** [review]

**Pull request merged**